packages/component-base/src/virtualizer-iron-list-adapter.js

```diff
@@ -178,6 +178,16 @@ export class IronListAdapter {
     }
   }
 
+  hostConnected() {
+    // Restore scroll position, which is reset when host is removed from DOM,
+    // since virtualizer doesn't re-render when adding it to the DOM again.
+    // If the scroll target isn't visible and its `offsetParent` is `null`, wait
+    // for the ResizeObserver to handle this case (hiding -> moving -> showing).
+    if (this.scrollTarget.offsetParent && this.scrollTarget.scrollTop !== this._scrollPosition) {
+      this.scrollTarget.scrollTop = this._scrollPosition;
+    }
```

Comment:
Can we now remove the previous implementation that used `ResizeObserver`?

https://github.com/vaadin/web-components/blob/0804c643439558500d217a6f0a3e22b50840c15b/packages/component-base/src/virtualizer-iron-list-adapter.js#L53-L65

Reply:
IIRC virtualizer doesn't render when the element is hidden, so this is probably still needed.

Reply:
Yes, it's still needed since it covers case of first moving the element and then showing it.